std/experimental/allocator/package.d

```diff
@@ -443,6 +449,21 @@ auto make(T, Allocator, A...)(auto ref Allocator alloc, auto ref A args)
         return emplace!T(m, args);
     else return emplace(cast(T*) m.ptr, args);
 }
+/// ditto
+auto make(T, Allocator, P, A...)(auto ref Allocator alloc, auto ref P outer, auto ref A args)
+    if (is(T == class) && isInnerClass!T && is(P : AliasSeq!(__traits(parent, T))[0]))
```

Comment:
The two specializations accept different parameters. So it's not possible to have just one function with a `static if` inside. Also, I cannot forward the second specialization to the first, as they have mutually exclusive constraints.